generate.go

```diff
@@ -45,13 +50,28 @@ type DependencyManager interface {
 	GenerateBillOfMaterials(dependencies ...postal.Dependency) []packit.BOMEntry
 }
 
-func Generate(dependencyManager DependencyManager, logger scribe.Emitter) packit.GenerateFunc {
+type EtcPasswdFileContentGetter func() (string, error)
+
+type DuringBuildPermissionsGetter struct {
+	get_etc_passwd_file_content EtcPasswdFileContentGetter
+}
+
+func NewDuringBuildPermissionsGetter(epfg EtcPasswdFileContentGetter) DuringBuildPermissionsGetter {
+	return DuringBuildPermissionsGetter{get_etc_passwd_file_content: epfg}
+}
+
+func Generate(dependencyManager DependencyManager, logger scribe.Emitter, duringBuildPermissionsGetter DuringBuildPermissionsGetter) packit.GenerateFunc {
```

Comment:
I'm wondering if it would be simpler/cleaner to 

1) Just pass in the UID and GUID in a structure to Generate. The tests would just hard code those values when testing Generate
2) Make the function to get the UID and GUID from the password file take a filename.  We could then add tests which used a dummy password file(s) in the test directories to validate the functionality of reading extracting the UIDs and GUIDs
3) Update main.go to simply call the function to get the UID and GUID with /etc/passwd and call Generate with the structure returned?

Reply:
I just pushed the changes, feel free to have a look and give me more feedback :) @mhdawson @thitch97